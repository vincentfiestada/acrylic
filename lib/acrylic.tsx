/**
 * Re-export all components in one place
 */
export { default as IProps } from "./components/IProps";
export { default as IEventProps } from "./components/IEventProps";

/**
 * Layout Components
 */
export { default as ILayoutProps } from "./components/layout/ILayoutProps";
export { default as Section } from "./components/layout/Section";
export { default as Column } from "./components/layout/Column";
export { default as Row } from "./components/layout/Row";
export { default as Alert } from "./components/layout/Alert";
export { default as Toolbar } from "./components/layout/Toolbar";

/**
 * Typography Components
 */
export { default as ITextProps } from "./components/typography/ITextProps";
export { default as Title } from "./components/typography/Title";
export { default as Text } from "./components/typography/Text";
export { default as Icon } from "./components/typography/Icon";

/**
 * Control Components
 */
export { default as Button } from "./components/controls/Button";
export { default as IconButton } from "./components/controls/IconButton";

/**
 * Variants
 */
export { default as Variant } from "./components/Variant";
