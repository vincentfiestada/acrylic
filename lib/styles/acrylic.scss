@import "normalize.css";
@import "typography";
@import "controls";
