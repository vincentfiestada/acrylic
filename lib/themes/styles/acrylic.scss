@import "normalize.css";
@import "layout";
@import "typography";
@import "icons";
@import "controls";
@import "animation";
