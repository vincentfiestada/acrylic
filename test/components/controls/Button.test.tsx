import * as React from "react";
import { mount } from "enzyme";
import { Button } from "../../../lib/acrylic";

describe("<Button />", () => {
	it("Should match the latest snapshots", () => {
		expect(
			mount(
				<React.Fragment>
					<Button>Button</Button>
					<Button variant="primary">Primary</Button>
					<Button variant="success">Success</Button>
					<Button variant="warning">Warning</Button>
					<Button variant="danger">Danger</Button>
					<Button disabled>Disabled</Button>
				</React.Fragment>
			)
		).toMatchSnapshot();
	});
	it("Should support the 'click' event", () => {
		const spy = jest.fn();
		const buttonComponent = mount(<Button onClick={spy} />);
		buttonComponent.simulate("click");
		expect(spy).toBeCalled();
	});
});
