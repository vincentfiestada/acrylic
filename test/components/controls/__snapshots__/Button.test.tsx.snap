// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should match the latest snapshots 1`] = `
Array [
  <Button>
    <button
      className="acr-btn"
    >
      Button
    </button>
  </Button>,
  <Button
    variant="primary"
  >
    <button
      className="acr-btn acr-variant-primary"
    >
      Primary
    </button>
  </Button>,
  <Button
    variant="success"
  >
    <button
      className="acr-btn acr-variant-success"
    >
      Success
    </button>
  </Button>,
  <Button
    variant="info"
  >
    <button
      className="acr-btn acr-variant-info"
    >
      Info
    </button>
  </Button>,
  <Button
    variant="warning"
  >
    <button
      className="acr-btn acr-variant-warning"
    >
      Warning
    </button>
  </Button>,
  <Button
    variant="danger"
  >
    <button
      className="acr-btn acr-variant-danger"
    >
      Danger
    </button>
  </Button>,
  <Button
    disabled={true}
  >
    <button
      className="acr-btn"
      disabled={true}
    >
      Disabled
    </button>
  </Button>,
]
`;
